proto/buf/alpha/registry/v1alpha1/plugin_curation.proto

```diff
@@ -114,6 +116,49 @@ service PluginCurationService {
   rpc GetLatestCuratedPlugin(GetLatestCuratedPluginRequest) returns (GetLatestCuratedPluginResponse);
 }
 
+// PluginCodeGenerationService generates code using remote plugins.
+service PluginCodeGenerationService {
+  // GenerateCode generates code using the specified remote plugins.
+  rpc GenerateCode(GenerateCodeRequest) returns (GenerateCodeResponse);
+}
+
+message GenerateCodeRequest {
+  // The image to run plugins against to generate the desired file outputs.
+  //
+  // All image files that are not imports and not well-known types will be generated.
+  // If you want to filter what files are generated, modify the image.
+  // If you want to include imports, set include_imports.
+  buf.alpha.image.v1.Image image = 1;
+  repeated PluginGenerationRequest requests = 2;
+  // Include imports from the Image in generation.
+  bool include_imports = 3;
+  // Include Well-Known Types from the Image in generation.
+  //
+  // include_imports must be set if include_well_known_types is set.
+  bool include_well_known_types = 4;
+}
+
+message GenerateCodeResponse {
+  // Responses from each plugin execution.
+  // The order of each response matches the order in the GenerateCodeRequest.
+  repeated PluginGenerationResponse responses = 1;
+}
+
+// Request for performing code generation using the specified plugin.
+message PluginGenerationRequest {
+  // The plugin to execute.
+  CuratedPluginReference plugin = 1;
+  // The options to pass to the plugin. These will
+  // be merged into a single, comma-separated string.
+  repeated string options = 2;
+}
+
+// Response from code generation for a given plugin.
+message PluginGenerationResponse {
+  // Generated code from the plugin.
+  google.protobuf.compiler.CodeGeneratorResponse response = 1;
```

Comment:
This felt better to break into a separate message type for future expansion - for example if we want to return stdout/stderr or any other fields. The previous `GeneratePluginsResponse` for example returned `repeated RuntimeLibrary` but it wasn't clear which plugin each library referred to. This allows us to group returned data by plugin.

Reply:
Indexing once is an advantage but if you look here: https://github.com/bufbuild/buf/blob/3863d2fc5e0e96acd7079c7d82d5ce86a3fa7c29/proto/buf/alpha/registry/v1alpha1/generate.proto#L92-L98

Let's say you were generating code with multiple plugins as part of one request. It wasn't possible in the previous API to know which plugin each RuntimeLibrary was associated with (a single plugin can return 0-N RuntimeLibrary messages).

Reply:
Got it, makes sense to me.